src/components/NewWorkspaceModal.js

```diff
@@ -111,7 +111,12 @@ const NewWorkspaceModal = withDisplayName('NewWorkspaceModal', ({
         }],
         async () => {
           const workspace = await Ajax().Workspaces.create(body)
-          Ajax().Metrics.captureEvent(Events.workspaceCreate, extractWorkspaceDetails(
+          // Only invoke Leo if we are within an Azure Workspace
+          if (!workspace.googleProject) {
+            await Ajax().Leo.create(workspace.name, workspace.id)
+          }
+          // TODO: Probably should add a metric for the Leo creation too
+          await Ajax().Metrics.captureEvent(Events.workspaceCreate, extractWorkspaceDetails(
```

Comment:
No need to add the await. This is a fire and forget sort of thing, and we usually do not await `captureEvent`.

I would suggest you remove the TODO. Either add the metric, or create a ticket for someone's backlog, but having a TODO here isn't particularly helpful.